### ISSUE TITLE
Rules not being merged

### ISSUE THREAD

**Issue by Dylan-Chapman** [clean-css/clean-css#713]
Using through: gulp-minify-css@1.2.2 (has a dependency clean-css@3.4.8)
Options: Doesn't matter (default produces issue)
OS: Windows 7

I'm using Lost for my site's grid system. I also decided to create Bootstrap-esque classes using it.

This is the code that clean-css is operating on:

``` css
.col-xs-1 {
    -webkit-box-flex: 0;
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    width: calc(99.99% * 1/12 - 13.75px)
}

.col-xs-1:nth-child(n) {
    margin-right: 15px
}

.col-xs-1:last-child,.col-xs-1:nth-child(12n) {
    margin-right: 0
}

.col-xs-2 {
    -webkit-box-flex: 0;
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    width: calc(99.99% * 2/12 - 12.5px)
}

.col-xs-2:nth-child(n) {
    margin-right: 15px
}

.col-xs-2:last-child, .col-xs-2:nth-child(6n) {
    margin-right: 0
}

/* Goes through .col-xs-12 */
```

However, it does nothing with this code.

I would expect the output to be something akin to:

``` css
.col-xs-1, .col-xs-2 {
    -webkit-box-flex: 0;
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
}

.col-xs-1 {
    width: calc(99.99% * 1/12 - 13.75px)
}

.col-xs-2 {
    width: calc(99.99% * 2/12 - 12.5px)
}

.col-xs-1:nth-child(n), .col-xs-2:nth-child(n) {
    margin-right: 15px
}

.col-xs-1:last-child,.col-xs-1:nth-child(12n), .col-xs-2:last-child, .col-xs-2:nth-child(6n) {
    margin-right: 0
}
```

Is there some reason these aren't being merged?

**Comment by jakubpawlowicz** [comment]
In this very case we are just too conservative. Let's fix it.

**Comment by superpoincare** [comment]
I am having a similar issue, although somewhat simpler: 

```
.entry-content blockquote,
.comment-content blockquote {
	margin-bottom: 24px;
	margin-bottom: 1.714285714rem;
	padding: 24px;
	padding: 1.714285714rem;
	font-style: italic;
}

.
.
.

.entry-content blockquote,
.comment-content blockquote {
	border-left: 1px solid #d9d9d9;
	border-left: 1px solid rgba(68, 68, 68, 0.2);
	color: #696969;
	color: rgba(68, 68, 68, 0.8);
	font-style: normal;
}
```

should be 

```
.entry-content blockquote,
.comment-content blockquote {
	margin-bottom: 24px;
	margin-bottom: 1.714285714rem;
	padding: 24px;
	padding: 1.714285714rem;
	border-left: 1px solid #d9d9d9;
	border-left: 1px solid rgba(68, 68, 68, 0.2);
	color: #696969;
	color: rgba(68, 68, 68, 0.8);
}
```

but instead I see them separated.

**Comment by jakubpawlowicz** [comment]
Depends on #623.

**Comment by matt3224** [comment]
I'm having a similar issue, whereby clean-css is not doing any merging. I created a test case here:

https://github.com/matt3224/remove-dupe-props-test

The out is the same, I would expect it to merge the rules and remove overridden properties like refresh-sf does.

Any help would be much appreciated

**Comment by jakubpawlowicz** [comment]
This issue depends on a proper implementation of merging for `flex` properties.

**Comment by matt3224** [comment]
refresh-sf use clean-css and have it working quite well from what i can see have you taken a look?

If i paste in:
```
.button {
    align-items: center;
    background-color: black;
    border: 1px solid black;
    border-radius: 4px;
    bottom: 0;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
    box-sizing: border-box;
    color: white;
    cursor: pointer;
    display: flex;
    flex: 1 1 auto;
    flex-direction: column;
    font-family: 'Arial', sans-serif;
    height: 50vh;
    left: 0;
    line-height: 1.3;
    margin: 1rem 1.5rem;
    min-height: 10rem;
    opacity: 0;
    overflow: auto;
    padding: 1rem;
    position: absolute;
    right: 0;
    text-align: center;
    transition: all 0.25s ease;
    top: 0;
    width: 100vw;
    z-index: 1;
}

.button {
    align-items: flex-end;
    background-color: red;
    border: 1px solid red;
    border-radius: 6px;
    bottom: 1rem;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.25);
    box-sizing: content-box;
    color: snow;
    cursor: grab;
    display: inline-flex;
    flex: 1 0 100%;
    flex-direction: row;
    font-family: 'Helvetica', 'Arial', sans-serif;
    height: 40vh;
    left: 1rem;
    line-height: 1.5;
    margin: 1rem 2rem;
    max-width: 40rem;
    min-height: 20rem;
    opacity: 0.8;
    overflow: visible;
    padding: 1.5rem;
    position: fixed;
    right: 1rem;
    text-align: start;
    transition: all 0.5s ease;
    top: 1rem;
    width: 90vw;
    z-index: 2;
}
```
it will give me this code (prettified for readability):
```
.button {
  align-items: flex-end;
  background-color: red;
  border: 1px solid red;
  border-radius: 6px;
  bottom: 1rem;
  box-shadow: 0 0 20px rgba(0, 0, 0, .25);
  box-sizing: content-box;
  color: snow;
  cursor: grab;
  display: inline-flex;
  flex: 1 0 100%;
  flex-direction: row;
  font-family: Helvetica, Arial, sans-serif;
  height: 40vh;
  left: 1rem;
  line-height: 1.5;
  margin: 1rem 2rem;
  max-width: 40rem;
  min-height: 20rem;
  opacity: .8;
  overflow: visible;
  padding: 1.5rem;
  position: fixed;
  right: 1rem;
  text-align: start;
  transition: all .5s ease;
  top: 1rem;
  width: 90vw;
  z-index: 2
}
```

**Comment by matt3224** [comment]
refresh-sf uses version 3 but it is not available on npm, where can i get this?

**Comment by jakubpawlowicz** [comment]
Won't fix as clean-css is going into [maintenance mode](https://github.com/clean-css/clean-css/discussions/1209).
